benchmarks/benchmarks/integrate.py

```diff
@@ -111,4 +111,5 @@ def time_quad_ctypes(self):
         quad(self.f_ctypes, 0, np.pi)
 
     def time_quad_cffi(self):
-        quad(self.f_cffi, 0, np.pi)
+        if cffi is not None:
+            quad(self.f_cffi, 0, np.pi)
```

Comment:
What kind of failure are you seeing? Normally `asv` does a good job of reporting a missing dependency as a proper (but not catastrophic) failure if the import is done properly (usually `try:` and `except: pass` or similar).

`asv` reporting a failure in the absence of a dependency sounds like the correct behavior, while adding shim code inside of a `time_..` function ir normally not desired.

If we want `cffi` to be present for the "standard" benchmark runs, we could commit it into `benchmarks/asv.conf.json` dependency `matrix`.

Reply:
There are tracebacks in the issue linked from the PR description.

Reply:
Looks "ok" to me--the dependency is missing so the benchmarking process moves on after printing the traceback.

Reply:
Also, was hesitant to click a google drive link--in any case, I don't see a reason to include the timing of `is not None` here when running benchmarks instead of just turning the dependency on if we really want it to be standard, and ignoring a few printed error lines if CFFI is not present.

Reply:
You'd actually get two bad side effects:

- when `cffi` is present, the timing is convoluted with an extra conditional check
- when `cffi` is absent, `asv` runs the benchmark as if nothing is missing, and reports timings for a 1-line conditional check instead of the function under benchmark, which is deceptive and less useful that the original (printed/handled) error message

Reply:
Can we do something like skipping a test if `cffi` is not present?

Reply:
Normally we'd just leave it alone. Making the existence of a benchmark conditional on the existence of an import doesn't seem worth the mess to me.

Reply:
Could be a useful upstream feature, though bringing the skip/xfail feature levels up to parity with something like `pytest`  may run into bandwidth issues for the small `asv` team (just Pauli that merges stuff these days I think?).